lib/block-supports/layout.php

```diff
@@ -51,24 +52,118 @@ function gutenberg_get_layout_style( $selector, $layout, $has_block_gap_support
 		$all_max_width_value  = wp_strip_all_tags( explode( ';', $all_max_width_value )[0] );
 		$wide_max_width_value = wp_strip_all_tags( explode( ';', $wide_max_width_value )[0] );
 
-		$style = '';
+		// Add universal styles for all default layouts.
+		// Add left align rule;
+		$class_names[] = $style_engine->add_style(
```

Comment:
❤️ 

This is a creative way of getting around the unique id. I wonder how we could marry it with some overarching and semi-permanent list of layout tokens (and their values, assuming they exist in theme.json)

Reply:
It would be good to centralise somewhere the different parts of the class name instead of having them added in an ad hoc way? 🤔

Reply:
> It would be good to centralise somewhere the different parts of the class name instead of having them added in an ad hoc way?

Yeah, I wonder how much we can get from a set of defaults + theme.json overrides.

Taking `wp-layout-default-content-size-${ suffix }` as an example, I'm seeing a few morphological units.

Assuming I knew nothing about the layout implementation, but am familiar with Gutenberg blocks:

1. `wp-layout`. This is familiar. I suppose it belongs to the block that uses this support feature. A reliable hook that will always be there.
2. `default`. This doesn't reveal much information. Is the default layout relate to the CSS rules that will be applied if I do nothing? Could this mean "fluid", or "flow"?
3. `content`. Stringing the morphemes together, I think we're talking about the layout of a piece of content, and because it's block supports, the content of the block.
4. `size` - I guess we're talking about "width", the value of which could be derived from a set of design rules, support by the wider system, or random (?)



For example, I'd find it something like the following reasonably readable:

```html
<div class="wp-layout wp-layout__flex wp-layout__width--large" />
```

Might be too BEM.

Still, there are probably some things we could borrow from BEM, [tailwind](https://tailwindcss.com/) or [cube](https://cube.fyi/composition.html#examples), which uses `data` attributes as selectors for exceptions. Very interesting!

Reply:
*Please* don't use data attributes. They're for holding data; class names are for this specific purpose.